### ISSUE TITLE
WiFi

### ISSUE THREAD

**Issue by drewwells** [EricSmekens/node-bluetooth-obd#26]
Can this be adapted to the wifi serial interface? 192.168.0.10:35000

**Comment by EricSmekens** [comment]
That would require quite some changes. Could be possible ofcourse, you just need to swap out the bluetooth-serial-connection with a wifi kind of module.

**Comment by drewwells** [comment]
I was thinking of switching out bluetooth with this. https://github.com/monteslu/udp-serial

Not sure if this package works, I've only got a successful connection to my elm327. Haven't been able to read or write to the port.

**Comment by arjunroychowdhury** [comment]
I've forked a version that uses TCP - I've tested it with obdsim so far. Seems to work fine. Work in progress. https://github.com/hsccorp/node-bluetooth-tcp-obd and an example at https://github.com/hsccorp/obd-tcp-test

My JS skills aren't up to snuff, so I'm sure I've done things that are not ideal. Happy to accept PRs and happy to get this merged with @EricSmekens library if he'd like - but I have no way to test if my changes affected the BT part (plus it will need better cleanup)

**Comment by EricSmekens** [comment]
Cool, I will try to have a look soon. Would be cool to merge this, and make a generic module that could work with bluetooth/tcp and serial. But time is an issue.

**Comment by EricSmekens** [comment]
@arjunroychowdhury Did you manage to test this on a simulator?
I am willing to spend time on this project again, but testing this requires qiute a lot of time, which demotivates the effort to put into this.

Would be nice to know what you think of this.

**Comment by arjunroychowdhury** [comment]
@EricSmekens, yes. I'm using it with the TCP fork of obdsim and a real mobile app
Summary of links:
a) My TCP fork of your excellent work: https://github.com/hsccorp/node-bluetooth-tcp-obd
b) TCP fork of obdsim that I am using to test https://github.com/oesmith/obdgpslogger
c) My sample mobile app that uses the TCP fork of your library https://github.com/hsccorp/obd-tcp-test
